### PR TITLE
[R2] Added object lifecycles docs

### PR DESCRIPTION
**Pull request by dcpena**
Added docs for object lifecycles. Addresses PCX-5458.

### CODE REVIEW

File: content/r2/objects/object-lifecycles.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+title: Object lifecycles
+pcx_content_type: how to
+---
+
+{{<beta heading="h1">}} Object lifecycles {{</beta>}}
+
+Object lifecycles affect how long objects uploaded to your bucket are kept.
+
+For example, you can create an object lifecycle rule to delete objects after 30 days, or you can set a rule to delete parts of multipart uploads.
```
Comment (harshal317) [inline content/r2/objects/object-lifecycles.md]: `delete parts of multipart uploads` isn't quite true. Maybe `or you can set a rule to abort multipart uploads after 30 days`

File: content/r2/objects/object-lifecycles.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+When you apply a rule to a bucket, the rule automatically applies to any new objects uploaded after the rule creation. For existing objects, the new rule may take a while to apply and will usually occur within 24 hours but can take longer based on the number of objects in the bucket.
```
Comment (harshal317) [inline content/r2/objects/object-lifecycles.md]: If we're specifying this in the behavior section is it worth having here as well?

File: content/r2/objects/object-lifecycles.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+## Behavior
+
+- Objects will typically be removed from a bucket within 24 hours of the `x-amz-expiration` value.
+- When a new lifecycle policy is applied, the new object's `x-amz-expiration` value immediately reflects the expiration based on the new rules, but existing objects may experience a delay. Most buckets will be transitioned within 24 hours but may take longer depending on bucket size. While objects are being migrated, you may see Object Lifecycle Management apply rules from the previous configuration.
```
Comment (harshal317) [inline content/r2/objects/object-lifecycles.md]: `the new object's` -> `newly uploaded objects`?

Also `Most buckets will be transitioned within 24 hours but may take longer depending on bucket size` -> `Most objects will be transitioned within 24 hours but may take longer depending on the number of objects in the bucket.`

**Review by kodster28 — APPROVED** [review]

File: content/r2/objects/object-lifecycles.md (hunk 3 above)
Comment (ferhatelmas) [inline content/r2/objects/object-lifecycles.md]:
```diff
- you may see Object Lifecycle Management apply rules from the previous configuration
+ you may see old applied rules from the previous configuration
```

File: content/r2/objects/object-lifecycles.md
Comment (ferhatelmas) [inline content/r2/objects/object-lifecycles.md:1]: we're using policy and rule interchangeably, might be a good idea to stick to one.

Comment (harshal317) [reply content/r2/objects/object-lifecycles.md:1]: I agree. I think a rule is fine to use to describe an individual element of the policy / configuration. We should use one of policy / configuration though.

File: content/r2/objects/object-lifecycles.md (hunk 1 above)
Comment (dcpena) [reply content/r2/objects/object-lifecycles.md]: ✔️

File: content/r2/objects/object-lifecycles.md (hunk 2 above)
Comment (dcpena) [reply content/r2/objects/object-lifecycles.md]: Removed because it's covered in behavior

File: content/r2/objects/object-lifecycles.md (hunk 3 above)
Comment (dcpena) [reply content/r2/objects/object-lifecycles.md]: ✔️

Comment (dcpena) [reply content/r2/objects/object-lifecycles.md]: ✔️

File: content/r2/objects/object-lifecycles.md
Comment (dcpena) [reply content/r2/objects/object-lifecycles.md:1]: Good catch. I swapped out the few instances of configuration for "policy" instead.

**Pull request merged**